appendix/single-sign-on.rst

```diff
@@ -0,0 +1,521 @@
+
+   **What does this all mean?**
+
+   It means there are many ways you could set up SSO—you
+   don’t need to follow this guide or even use Active Directory or Kerberos—but
+   if you don’t know what you’re doing,
+   you’re going to end up with a *massive* security hole.
+
+Getting Started
+---------------
+
+.. hint:: 😵 **Too busy to handle it on your own?**
+
+   We’ve got you covered.
+   Our experts offer custom-tailored workshops
+   to get your team up and running fast and with confidence.
+   `Just drop us a line <https://zammad.com/contact>`_!
+
+You will need:
+
+* a Microsoft Active Directory environment with
+
+  * root access
+  * support for AES 256-bit encryption
+
+* a Zammad host with
+
+  * root access
+  * a fully-qualified domain name (FQDN)
+
+* some familiarity with system administration (*e.g.,* Apache configuration)
+
+For best results, set up `LDAP integration`_
+to make sure your Active Directory and Zammad user accounts
+are always in sync.
+
+.. _LDAP integration: https://admin-docs.zammad.org/en/latest/system/integrations/ldap.html
+
+.. _sso-register-spn:
+
+Step 1: Configure Active Directory
+----------------------------------
+
+In the Kerberos authentication scheme,
+the **authentication server** (Active Directory)
+needs to maintain shared secrets with the **service** (Zammad).
+To make this possible, we need to register a **service principal name** (SPN)
+for Zammad on Active Directory.
+
+.. note:: These directions have been confirmed on Windows Server 2016.
+
+1a. Create a service account
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+You may use an existing service account if you have one.
+Admin privileges are not required; a normal user account will do.
+
+Select “This account supports Kerberos AES 256 bit encryption” under
+**Properties > Account > Account options**.
+
+1b. Register an SPN for Zammad
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+.. note:: Replace the following placeholders in the command below:
+
+   :``<zammad-host>``:              Zammad FQDN
+   :``<service-acct>``:             Service account logon name
+   :``<service-acct-pwd>``:         Service account password
+   :``<domain>``:                   Windows domain
+   :``<master-domain-controller>``: Master domain controller IP/FQDN
+
+.. code-block:: sh
+
+   $ setspn -s HTTP/<zammad-host> <service-acct>
+   $ ktpass /princ <service-acct>@<domain> \
+            /mapuser <service-acct> \
+            /crypto AES256-SHA1 \
+            /ptype KRB5_NT_PRINCIPAL \
+            /pass <service-acct-pwd> -SetPass +DumpSalt \
+            /target <master-domain-controller> \
+            /out zammad.keytab
+
+1c. Note the secret key and version number
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+The output of the command above contains important data for Step 2e below:
+
+.. code-block:: none
+
+   Using legacy password setting method
+   Failed to set property 'servicePrincipalName' to 'HTTP/<zammad-host>' on Dn 'CN=Zammad Service,DC=<domain>,DC=<tld>': 0x13.
```

Comment:
I replaced all of your `zammadsrv` / `THA` / `DEV` values here with more placeholders for the sake of anonymity.